src/evm/Readme.md

```diff
@@ -2,73 +2,71 @@
-and overriding account balances and storage slots.
+To perform simulations, the engine requires access to the EVM state, provided by the `EngineDatabaseInterface` trait.
+Two implementations of this trait are available:
+
+`PreCachedDB`: Preloads necessary data such as contract code, storage slots, and balances. This implementation uses
+Tycho
+Indexer as its data source and offers the most efficient simulation experience.
+
+`SimulationDB`: Queries an Ethereum node for required data and caches it for the duration of one block. It also supports
+mocking accounts.
+
+By default, `SimulationEngine` uses `PreCachedDB` for optimal efficiency. However, `SimulationDB` is available for
+testing or
+scenarios requiring dynamic queries.
 
 ## Overriding chain state
 
 Terms used in this chapter:
 
-- `engine` - an instance of `SimulationEngine`
-- `state` - an instance of `SimulationDB` that the `engine` is using
+- `engine` - An instance of `SimulationEngine`
+- `state` - An instance of `EngineDatabaseInterface` that the `engine` is using
 
 ### Default behaviour
 
-In a vanilla engine, if nothing is explicitly done to override a state, simulations happen as if they were executed on
-current block.
-
-Whenever some data needed by the engine is not present in the state, the state queries an Ethereum node for this data.
-This way the engine gets basic information about accounts (such as balance, nonce and code) as well as accounts'
-storage (values stored by smart contracts).
-
-These values are cached until `state.clear_temp_storage()` is called.
+In the default setup, simulations execute as if they were performed on the current block, unless explicitly overridden.
 
 ### Overriding an account
 
-Instead of querying account information or storage from a node, we can set it manually.
-
-This is done with `state.init_account()` method which takes the following arguments:
-
-- account address
-- `AccountInfo` instance
-- permanent storage mapping
-- `mocked` boolean
+Instead of relying on live queries for account data, you can manually configure account information using
+the `state.init_account()` method. This method accepts the following parameters:
 
-`AccountInfo` contains basic account information such as balance, nonce and code (in case of smart contract accounts).
+```
+address: Address,
+account: AccountInfo,
+permanent_storage: Option<HashMap<U256, U256>>,
+mocked: bool,
+```
 
-Permanent storage lets us set some storage slots. If a value of a certain storage slot is needed during simulation, the
-state will first check the permanent storage set here. Only if a slot is not set in the permanent storage, the state
-will query an Ethereum node (but see the next section).
+- `AccountInfo`: Contains core account details like balance, nonce, and code (for smart contract accounts).
+- `permanent_storage`: Allows setting predefined storage slots. When the simulation requires a specific storage slot,
+  the engine first checks this storage.
```

Comment:
Hmmm... I wouldn't say 'first checks' this storage. Below you directly contradict this and say the simulation overrides are used before these permanent ones are. Maybe we should word it along the lines of: during simulations, the engine will prioritise the slots listed here above those provided in the db/by the data source?